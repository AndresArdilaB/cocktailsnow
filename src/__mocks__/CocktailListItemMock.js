const CocktailListItemMock = {
  idDrink: '14378',
  strDrink: 'Lunch Box',
  strCategory: 'Beer',
  strAlcoholic: 'Alcoholic',
  strGlass: 'Pint glass',
  strInstructions: 'Fill a pint glass almost full with beer. Then fill the rest with orange juice (careful not to fill it to the top). Then take the shot of Amaretto and drop it in.',
  strInstructionsDE: 'Füllen Sie ein Pintglas fast voll mit Bier. Dann den Rest mit Orangensaft füllen (darauf achten, ihn nicht nach oben zu füllen). Dann nimm den Schuss Amaretto und lass ihn rein.',
  strDrinkThumb: 'https://www.thecocktaildb.com/images/media/drink/qywpvt1454512546.jpg',
  strIngredient1: 'Beer',
  strIngredient2: 'Amaretto',
  strIngredient3: 'Orange juice',
  strMeasure1: '3/4 bottle ',
  strMeasure2: '1 shot ',
  strMeasure3: '1 oz ',
  strCreativeCommonsConfirmed: 'No',
  dateModified: '2016-02-03 15:15:46',
};

export default CocktailListItemMock;
