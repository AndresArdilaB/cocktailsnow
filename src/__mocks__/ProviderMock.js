import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';
import { testState } from '../frontend/initialState';
import reducer from '../frontend/reducers';

const store = createStore(reducer, testState);
const history = createBrowserHistory();

const ProviderMock = ({ children }) => (
  <Provider store={store}>
    <Router history={history}>
      {children}
    </Router>
  </Provider>
);

export default ProviderMock;
