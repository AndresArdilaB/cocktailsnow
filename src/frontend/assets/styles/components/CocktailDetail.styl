.detail
  margin: 20px 0px

.title
  text-align: center
  width: 100%
  font-size: 18px

.detail-container
  display: flex
  flex-direction: row
  margin: 20px auto
  width: fit-content
  @media screen and (max-width: 1100px)
    width: 540px
  @media screen and (max-width: 600px)
    width: 280px
    flex-direction: column

.media
  display: flex
  justify-content: center
  border: solid 2px #fff
  border-radius: 5px
  margin-right: 10px
  width: fit-content
  // @media screen and (max-width: 600px)

.media img 
  max-width: 400px
  width: auto
  object-fit: cover
  @media screen and (max-width: 1100px)
    max-width: 265px

.text
  width: 400px
  @media screen and (max-width: 600px)
    width: 265px

.propierty
  display: flex
  flex-direction: row
  align-items: baseline

  h3
    margin-right: 5px

