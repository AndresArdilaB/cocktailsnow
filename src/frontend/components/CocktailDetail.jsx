/* eslint-disable prefer-template */
import React from 'react';
import '../assets/styles/components/CocktailDetail.styl';

const CocktailDetail = ({ data: { data, type } }) => {

  return (
    <section className='detail'>
      <div className='title'>
        {type === 'random' ? 'Random Cocktel' : 'Selected Cocktel'}
      </div>
      <div className='detail-container'>
        <div className='media'>
          <img src={data.strDrinkThumb} alt='Cocktail' />
        </div>
        <div className='text'>
          <div className='propierty' id={data.strDrink}>
            <h3>Name:</h3>
            <span>
              {data.strDrink}
            </span>
          </div>
          <div className='propierty'>
            <h3>Glass:</h3>
            <span>
              {data.strGlass}
            </span>
          </div>
          <div className='propierty'>
            <h3>Category:</h3>
            <p>
              {data.strCategory}
            </p>
          </div>
          <div>
            <h3>Ingredients:</h3>
            {data.strIngredient1 && <span>{data.strIngredient1}</span>}
            {data.strIngredient2 && <span>{', ' + data.strIngredient2}</span>}
            {data.strIngredient3 && <span>{', ' + data.strIngredient3}</span>}
            {data.strIngredient4 && <span>{', ' + data.strIngredient4}</span>}
            {data.strIngredient5 && <span>{', ' + data.strIngredient5}</span>}
            {data.strIngredient6 && <span>{', ' + data.strIngredient6}</span>}
            {data.strIngredient7 && <span>{', ' + data.strIngredient7}</span>}
            {data.strIngredient8 && <span>{', ' + data.strIngredient8}</span>}
            {data.strIngredient9 && <span>{', ' + data.strIngredient9}</span>}
            {data.strIngredient10 && <span>{', ' + data.strIngredient10}</span>}
          </div>
          <div>
            <h3>Instructions:</h3>
            <p>
              {data.strInstructions}
            </p>
          </div>
        </div>
      </div>
    </section>
  );
};

export default CocktailDetail;
