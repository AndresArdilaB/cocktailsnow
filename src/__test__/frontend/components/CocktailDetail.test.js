/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import { mount, shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import ProviderMock from '../../../__mocks__/ProviderMock';
import CocktailListItemMock from '../../../__mocks__/CocktailListItemMock';
import CocktailDetail from '../../../frontend/components/CocktailDetail';

describe('<CocktailDetail />', () => {

  test('Component render', () => {
    const cocktailDetail = shallow(
      <ProviderMock>
        <CocktailDetail />
      </ProviderMock>,
    );

    expect(cocktailDetail.length).toEqual(1);
  });

  test('Test detail', () => {
    const wrapper = mount(
      <ProviderMock>
        <CocktailDetail
          data={{ data: CocktailListItemMock, type: 'Detail' }}
        />
      </ProviderMock>,
    );

    expect(wrapper.find('[id="Lunch Box"]')).toHaveLength(1);
  });

});

describe('CocktailDetail Snapshot', () => {
  test('Testing render component UI', () => {
    const cocktailDetail = create(
      <ProviderMock>
        <CocktailDetail
          data={{ data: CocktailListItemMock, type: 'Detail' }}
        />
      </ProviderMock>,
    );

    expect(cocktailDetail.toJSON()).toMatchSnapshot();
  });
});
