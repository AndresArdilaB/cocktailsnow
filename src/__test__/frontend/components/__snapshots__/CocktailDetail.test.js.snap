// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CocktailDetail Snapshot Testing render component UI 1`] = `
<section
  className="detail"
>
  <div
    className="title"
  >
    Selected Cocktel
  </div>
  <div
    className="detail-container"
  >
    <div
      className="media"
    >
      <img
        alt="Cocktail"
        src="https://www.thecocktaildb.com/images/media/drink/qywpvt1454512546.jpg"
      />
    </div>
    <div
      className="text"
    >
      <div
        className="propierty"
        id="Lunch Box"
      >
        <h3>
          Name:
        </h3>
        <span>
          Lunch Box
        </span>
      </div>
      <div
        className="propierty"
      >
        <h3>
          Glass:
        </h3>
        <span>
          Pint glass
        </span>
      </div>
      <div
        className="propierty"
      >
        <h3>
          Category:
        </h3>
        <p>
          Beer
        </p>
      </div>
      <div>
        <h3>
          Ingredients:
        </h3>
        <span>
          Beer
        </span>
        <span>
          , Amaretto
        </span>
        <span>
          , Orange juice
        </span>
      </div>
      <div>
        <h3>
          Instructions:
        </h3>
        <p>
          Fill a pint glass almost full with beer. Then fill the rest with orange juice (careful not to fill it to the top). Then take the shot of Amaretto and drop it in.
        </p>
      </div>
    </div>
  </div>
</section>
`;
